### PR TITLE
Add Estimator Doc

### PR DESCRIPTION
**Pull request by cyita**

### CODE REVIEW

**Comment by qiuxin2012** [comment]
also add link to docs/mkdocs.yml

**Comment by cyita** [comment]
Done

File: docs/docs/APIGuide/PipelineAPI/estimator.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
+## Estimator
+
+
+`Estimator` supports the training and evaluation of BigDL models. It wraps a model, and provide a uniform training, evaluation, or prediction operation on both localhost and distributed spark environment.
```
Comment (qiuxin2012) [inline docs/docs/APIGuide/PipelineAPI/estimator.md]: not only bigdl model， but also keras-like model and pytorch model.

File: docs/docs/APIGuide/PipelineAPI/estimator.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+**Scala:**
+
+```scala
+val estimator = Estimator[T: ClassTag](
+      model: Module[T], 
+      optimMethods: Map[String, OptimMethod[T]] = Map(), 
+      modelDir: Option[String] = None)
+
+val estimator = Estimator[T: ClassTag](
```
Comment (qiuxin2012) [inline docs/docs/APIGuide/PipelineAPI/estimator.md]: split this.

File: docs/docs/APIGuide/PipelineAPI/estimator.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+```python
+estimator.train(train_set, criterion, end_trigger, checkpoint_trigger,
+              validation_set, validation_method, batch_size)
+```
+`train_set`: training dataset in type of FeatureSet[Sample[T]].\
+`criterion`: loss function.\
+`end_trigger`: when to finish the training.\
+`checkpoint_trigger`: how often to save a checkpoint and evaluate the model.\
+`validation_set`: validation dataset in type of FeatureSet[Sample[T]].\
+`validation_method`: a set of validationMethod.
```
Comment (qiuxin2012) [inline docs/docs/APIGuide/PipelineAPI/estimator.md]: br

File: docs/docs/APIGuide/PipelineAPI/estimator.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+```python
+estimator.train_imagefeature(self, train_set, criterion, end_trigger, checkpoint_trigger,
+                           validation_set, validation_method, batch_size)
+```
+`train_set`: training dataset in type of FeatureSet[ImageFeature].\
+`criterion`: loss function.\
+`end_trigger`: when to finish the training.\
+`checkpoint_trigger`: how often to save a checkpoint and evaluate the model.\
+`validation_set`: validation dataset in type of FeatureSet[ImageFeature].\
+`validation_method`: a set of validationMethod.
```
Comment (qiuxin2012) [inline docs/docs/APIGuide/PipelineAPI/estimator.md]: br

File: docs/docs/APIGuide/PipelineAPI/estimator.md (hunk 2 above)
Comment (qiuxin2012) [reply docs/docs/APIGuide/PipelineAPI/estimator.md]: also the parameter below.

**Comment by qiuxin2012** [comment]
please also backport to branch-0.6
